src/ServiceControl.Infrastructure.RavenDB/Extensions.cs

```diff
@@ -16,5 +19,21 @@ public static void Query<TState>(this DocumentDatabase db, string index, IndexQu
                 onItem(doc, state);
             }
         }
+
+        public static void ThrowWhenIndexErrors(this EmbeddableDocumentStore documentStore)
```

Comment:
Strictly speaking, I think we should move this part into a dedicated PR since the custom checks provides value on its own and for me this one is the most controversial change. But if other reviewers are fine with this change I'm OK to leave it here

Reply:
I think these changes are orthogonal and not really about reporting due the the blocking nature of throwing an exception. And w if we move this change out we can merge the custom check things and have a focused discussion on the potential edge cases about this change